### PR TITLE
NEWSPLT-118 Plus a whole lot more

### PR DESCRIPTION
**Pull request by drewwills**
This PR includes doodelicious' fine work on NEWSPLT-118:  Bootstrap-ize the UI.

  - https://issues.jasig.org/browse/NEWSPLT-118

It also includes...
* Some fixes to the original PR in its original scope
* Corrections to regressions introduced by the original PR vis-a-vis NRP's newest features:  infinite scroll & full story
* Corrections to UI strings in a few places
* COMPLETE REMOVAL of *-jQM jsp files and the ViewSelector technology
* COMPLETE REMOVAL of the single-feed-news portlet
* Addition of a Whitelist feature that allows admins to control which predefined feeds appear within which portlet-definitions

There are some major changes in this list.  Here is a brief explanation...

## Mobile Support ##

In the past, we (Apereo portlet developers) had mobile support strategy based on providing a different UI for mobile device users.  That paradigm has been supplanted by Responsive Design.   NEWSPLT-118 is all about 'Bootstrapizing' the portlet.  We're better off, going forward, handling mobile support in this way exclusively.  (Also see 'JSP Atrophy' below)

## single-feed-news ##

The reasons this portlet was added to the project no longer applies.  We added it because...
* We wanted to divide many feeds into different portlets appearing different places in the layout(s)
* We wanted to be able to do an initdb and see content in the portlet(s) right away (wanted to move away from empty portlets that require someone to fill them with content in the UI)

As far as the latter item -- we didn't have import/export in NRP when we created single-feed-news;  now we do.

Regarding the former item -- there is a better way to handle this need.  This PR adds (essentially) the same Whitelist strategy that Announcements uses:  the admin can configure a multi-valued preference of regex expressions that defines which predefined feeds appear in the portlet.  (The fname of a predefined feed must match at least one expression or it will not appear.)

(Incidentally, the 'news' portlet also includes the ability to disable the EDIT interface.)

## JSP Atrophy ##

In the tip of master right now, an uncommon amount of technical debt has accumulated in the JSPs.  The portlet is more & more JavaScript as time marches on, but rarely do all the JSPs get the maintenance they need when the core newsreader JS evolves.  There are many atrophied features in the JSPs.  I don't think a strategy of maintaining 3 different JSPs for each logical screen is sustainable.  We need to shift to maintaining 1 JSP for each purpose that handles all devices & configurations.

### CODE REVIEW

File: src/main/java/org/jasig/portlet/newsreader/mvc/AbstractNewsController.java [hunk 1]
```diff
@@ -50,16 +101,18 @@ public boolean isAdmin(PortletRequest req) {
...
+    }
 
     @ModelAttribute("supportsEdit")
     public boolean supportsEdit(PortletRequest req) {
-        return req.isPortletModeAllowed(PortletMode.EDIT);
-    }    
+        final PortletPreferences prefs = req.getPreferences();
+        final String allowEdit = prefs.getValue(ALLOW_EDIT_PREFERENCE, "true");
+        return Boolean.parseBoolean(allowEdit);
+    }
```
Comment (jameswennmacher) [inline src/main/java/org/jasig/portlet/newsreader/mvc/AbstractNewsController.java]: This is OK but is technically an API change and should probably bump at least the minor version # in the pom.xml.

File: src/main/java/org/jasig/portlet/newsreader/mvc/EmptyView.java [hunk 2]
```diff
@@ -77,17 +75,6 @@ public void setEncoding(JsonEncoding encoding) {
...
-     * Indicates whether the JSON output by this view should be prefixed with "{@code {} &&}". Default is false.
-     *
-     * <p> Prefixing the JSON string in this manner is used to help prevent JSON Hijacking. The prefix renders the string
-     * syntactically invalid as a script so that it cannot be hijacked. This prefix does not affect the evaluation of JSON,
-     * but if JSON validation is performed on the string, the prefix would need to be ignored.
-     */
-    public void setPrefixJson(boolean prefixJson) {
-        this.prefixJson = prefixJson;
-    }
-
```
Comment (jameswennmacher) [inline src/main/java/org/jasig/portlet/newsreader/mvc/EmptyView.java]: Are we saying that JSON Hijacking is not an issue anymore with modern browsers?  I saw a stack overflow comment suggesting that.

File: src/main/java/org/jasig/portlet/newsreader/service/Whitelist.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ * @author drewwills
+ */
+public class Whitelist<T> {
+
+    /**
+     * A multi-valued portlet preference that allows administrators to restrict
+     * which PredefinedNewsDefinitions will appear in a particular portlet
+     * definition.  In other words, you can publish the 'news' portlet multiple
+     * times and divide up the pre-defined feeds any way you like.  The default
+     * value for this preference is empty;  if the runtime value is is empty,
```
Comment (jameswennmacher) [inline src/main/java/org/jasig/portlet/newsreader/service/Whitelist.java]: minor - double 'is' in comment.

**Comment by jameswennmacher** [comment]
+1 Removal of unused JSPs reduces complexity and clutter.   

I do think the major version should be bumped because of the removal of single-news-feed and incorporation of the whitelist portlet preference.

File: src/main/java/org/jasig/portlet/newsreader/mvc/AbstractNewsController.java (hunk 1 above)
Comment (drewwills) [inline src/main/java/org/jasig/portlet/newsreader/mvc/AbstractNewsController.java]: @jameswennmacher -- I meant to cover this point in the writeup and forgot:  this is a new major version, 4.0.0.  The pom has already been bumped -- with earlier changes -- but there has been no release yet.

File: src/main/java/org/jasig/portlet/newsreader/mvc/EmptyView.java (hunk 2 above)
Comment (drewwills) [inline src/main/java/org/jasig/portlet/newsreader/mvc/EmptyView.java]: I'm not saying one way or another, but the `prefixJson ` variable was unused, so I removed it.

**Pull request merged**